### PR TITLE
Add custom writer for area

### PR DESCRIPTION
**Pull request by Brookke**
This is needed in order to support https://github.com/pterm/pterm/issues/261

### CODE REVIEW

File: area.go [hunk 1]
```diff
@@ -36,7 +45,7 @@ func (area *Area) Update(content string) {
 		}
 	} else {
 		for _, line := range lines {
-			fmt.Println(line)
+			fmt.Fprintln(area.writer, line)
```
Comment (Brookke) [inline area.go:51]: Due to the comments on https://github.com/atomicgo/cursor/pull/6 I assume this wouldn't work on Windows, so I have only added it to the non windows portion? Is this a correct assumption?

**Comment by MarvinJWendt** [comment]
Hi @Brookke, I just saw this PR again, is it done? It's currently marked as a draft, so I didn't review it :)

**Review by lammel — COMMENTED** [review]
Looks good to me and should allow flexible use on any writer

File: area.go [hunk 2]
```diff
@@ -10,11 +12,21 @@ import (
...
 // NewArea returns a new Area.
 func NewArea() Area {
-	return Area{}
+	return Area{
+		writer: os.Stdout,
+	}
+}
+
+// WithCustomWriter sets the custom writer
+func (area *Area) WithCustomWriter(writer io.Writer) *Area {
```
[the remark this answers is not in the document]
Comment (MarvinJWendt) [reply area.go:26]: Hi, thanks for taking a look at it. PTerm uses a wrapper of the `Area` here, so the method names don't have to be the same. But I also think `WithWriter` sounds cleaner.

**Pull request merged**